### PR TITLE
service controller: only sync LB node pools when relevant fields change

### PR DESCRIPTION
**Pull request by andrewsykim**
Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
In https://github.com/kubernetes/kubernetes/pull/81185 we added node event handlers in service controller so that LB node pools can be reconciled as soon as we received events for nodes. This PR optimizes the event handler so that we only run a sync if relevant node fields for the LB changes. 

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
service controller: only sync LB node pools when relevant fields in Node changes
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by andrewsykim** [comment]
cc @timoreimann

File: pkg/controller/service/controller.go [hunk 1]
```diff
@@ -649,6 +663,38 @@ func getNodeConditionPredicate() NodeConditionPredicate {
...
+	if !reflect.DeepEqual(oldNode.Labels, newNode.Labels) {
+		return true
+	}
+
+	oldNodeReadyStatus := nodeReadyConditionStatus(oldNode)
+	newNodeReadyStatus := nodeReadyConditionStatus(newNode)
+
+	// ready conditions unknown for both nodes, skip reconcile
+	if oldNodeReadyStatus == "" && newNodeReadyStatus == "" {
+		return false
```
Comment (MrHohn) [inline pkg/controller/service/controller.go:680]: nit: isn't this the same as `return oldNodeReadyStatus != newNodeReadyStatus`?

**Review by MrHohn — APPROVED** [review]
Thanks Andrew, this looks reasonable to me.

File: pkg/controller/service/controller.go (hunk 1 above)
Comment (wojtek-t) [reply pkg/controller/service/controller.go:680]: +1

**Review by wojtek-t — COMMENTED** [review]
LGTM from scalability/performance POV.

File: pkg/controller/service/controller.go (hunk 1 above)
Comment (andrewsykim) [reply pkg/controller/service/controller.go:680]: fixed

**Review by MrHohn — APPROVED** [review]
/lgtm

**Pull request merged**